### ISSUE TITLE
[Feature Request]: Expand the "Company" based seperation of information

### ISSUE THREAD

**Issue by pieterhouwen** [grokability/snipe-it#14709]
### Is your feature request related to a problem? Please describe.

We plan to use SnipeIT for multiple companies but as I currently understand the current architecture has a big limitation in terms of seperation of data. It is possible to group users to different companies but all the other information is available to all companies that are using the system.

### Describe the solution you'd like

When Full Company Support is enabled in the Admin Settings, users can only see the assets, locations, categories, accessories, components etc etc etc that are created in their own company. When logging in as (Super) Admin you can see all items of all companies. 

Users of Company A should have no rights to view, for example, Asset Models, License categories or Departments of Company B and should not be hindered because they want to create an asset category which has the same name and type as a category created by Company C

**Comment by pieterhouwen** [comment]
@snipe Any ideas/insights on this question?

**Comment by lcc-garry** [comment]
@snipe I completely agree that this is a much needed issue/feature. I would also like to recommend that a user can have the ability to have permissions in multiple companies. This would involve modifying the permissions module to be granular per company. That way you could assign these permissions to groups, and then the groups to the users.
